src/selectors/url-state.js

```diff
@@ -257,6 +257,33 @@ export const getFormattedMetaInfoString: Selector<string | null> = (state) => {
   return formatMetaInfoString(profile.meta);
 };
 
+/** Does the profile have implementation data? */
+export const getProfileUsesFrameImplementation: Selector<boolean> = (state) => {
+  const { profile } = state.profileView;
+  if (!profile) {
+    return true;
+  }
+  return profile.meta.doesNotUseFrameImplementation !== true;
+};
+
+/** Should the "Look up the function name on Searchfox" menu entry be hidden? */
+export const getShouldDisplaySearchfox: Selector<boolean> = (state) => {
+  const { profile } = state.profileView;
+  if (!profile) {
+    return true;
+  }
+  return profile.meta.sourceCodeIsNotOnSearchfox !== true;
+};
+
+/* Hide the stack type of frames in context menus? */
+export const getProfileUsesMultipleStackTypes: Selector<boolean> = (state) => {
+  const { profile } = state.profileView;
+  if (!profile) {
+    return true;
+  }
+  return profile.meta.usesOnlyOneStackType !== true;
+};
+
```

Comment:
Oh, I only noticed this just now: `src/selectors/url-state.js` is not the right place for these selectors, because they don't access the URL state. They only access `state.profileView`. So they should be moved to `src/selectors/profile.js`.
This includes `getFormattedMetaInfoString`, which was already in the wrong place. Could you move all four functions to `profile.js`?

Reply:
Of course and you guessed correctly, why I put it in the wrong place

Reply:
> This includes getFormattedMetaInfoString, which was already in the wrong place. Could you move all four functions to profile.js?

This leads to a circular dependency

Reply:
Oh, weird, I'll take a look.

Reply:
Ah, I see, we'd need to move `getProfileNameWithDefault` over as well, because it makes use of `getFormattedMetaInfoString`. Then let's not worry about it, it was an existing issue anyway.